src/lib/service/learnerRecordAPI/workers/test/workerTestUtils.ts

```diff
@@ -0,0 +1,194 @@
+export const testDateAsStr = '2020-01-01T00:00:00'
+
+export const testUser = new User('100', 'TestUser', '', [], '')
+export const testCourse = new Course('TEST001')
+
+// Course data
+
+export function getCourseWithOneRequiredModule() {
+	const module = getRequiredModule()
+	return genericCourse([module])
+}
+
+export function getCourseWithTwoRequiredModules() {
+	const course = getCourseWithOneRequiredModule()
+	course.modules.push(getRequiredModule())
+	return course
+}
+
+export function getCourseWithOneOptionalModule() {
+	const module = getOptionalModule()
+	return genericCourse([module])
+}
+
+export function getCourseWithTwoOptionalModules() {
+	const course = getCourseWithOneOptionalModule()
+	course.modules.push(getOptionalModule())
+}
+
+/**
+ * - module [0] is opt
+ * - module [1] is required
+ */
+export function getCourseWithMixedModules() {
+	const course = getCourseWithOneOptionalModule()
+	course.modules.push(getRequiredModule())
+	return course
+}
+
+export function genericCourse(modules: Module[]) {
+	const courseId = 'course ' + Math.random().toString()
+	const course = new Course(courseId)
+	course.modules = modules
+	return course
+}
+
+// Event data
+
+export function genericEvent() {
+	const eventId = 'event ' + Math.random().toString()
+	return new Event(new Date(), new Date(), [new Date()], 'Location', 10, 10, 'ACTIVE', eventId)
+}
+
+// Course record data
+
+export function getCourseRecordWithOneModuleRecord(
+	courseId: string,
+	courseState: RecordState,
+	moduleId: string,
+	moduleState: RecordState
+) {
+	const moduleRecord = createModuleRecord(moduleId, courseId, moduleState)
+	const courseRecord = genericCourseRecord(courseId, [moduleRecord])
+	courseRecord.state = courseState
+	return courseRecord
+}
+
+export function getCourseRecordWithTwoModuleRecords(
+	courseId: string,
+	courseState: RecordState,
+	moduleId: string,
+	moduleState: RecordState,
+	moduleTwoId: string,
+	moduleTwoState: RecordState
+) {
+	const moduleRecord = createModuleRecord(moduleId, courseId, moduleState)
+	const moduleRecordTwo = createModuleRecord(moduleTwoId, courseId, moduleTwoState)
+	const courseRecord = genericCourseRecord(courseId, [moduleRecord, moduleRecordTwo])
+	courseRecord.state = courseState
+	return courseRecord
+}
+
+export function genericCourseRecord(courseId: string, moduleRecords: ModuleRecord[]) {
+	return new CourseRecord(courseId, testUser.id, undefined, moduleRecords, 'Test course', false)
+}
+
+// Module data
+
+export function getRequiredModule() {
+	return genericModule(false)
+}
+
+export function getOptionalModule() {
+	return genericModule(true)
+}
+
+export function genericModule(optional: boolean) {
+	const moduleId = 'module ' + Math.random().toString()
+	const mod = new Module(moduleId, 'elearning')
+	mod.optional = optional
+	return mod
+}
+
+// Module record data
+
+export function createModuleRecord(moduleId: string, courseId: string, state: RecordState) {
+	const id = Math.random()
```

Comment:
Probably irrelevant because `id` isn't really being tested here, but since this is a test, `id` should be predictable and not randomised.